src/wallet/bdb.cpp

```diff
@@ -824,3 +824,50 @@ std::unique_ptr<DatabaseBatch> BerkeleyDatabase::MakeBatch(const char* mode, boo
 {
     return MakeUnique<BerkeleyBatch>(*this, mode, flush_on_close);
 }
+
+std::unique_ptr<BerkeleyDatabase> MakeBerkeleyDatabase(const fs::path& path, const DatabaseOptions& options, DatabaseStatus& status, bilingual_str& error)
+{
+    status = DatabaseStatus::SUCCESS;
```

Comment:
I think the non-BDB specific parts of this should be part of `MakeDatabase`. It would make this easier to handle in #19077